cmd/daemon.go

```diff
@@ -15,14 +21,62 @@ func NewDaemonCmd(global *globalFlags) *cobra.Command {
 		Use:   "daemon",
 		Short: "Run the crypta daemon",
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return c.Run(args)
+			if !c.detached {
+				return cmd.Help()
+			}
```

Comment:
Since the behavior of the command is different according to the flag, I would consider creating a separate subcommand like `run` and convert the `daemon` to just the group of the other three subcommands.

Reply:
The "nice thing" about this implementation is that the flag is hidden, and the daemon "main command" just returns the help message of the two sub commands if called without the flag. This way it is clear that the daemon needs to be started and stopped using the two sub commands.

Of course, if we want to support a separate `run` command that should also be executable by the user as well (i.e. not hidden), it makes sense to have it visible. But usually I would prefer to only have one way to start/stop something as it makes the interface easier to understand. Supporting both ways (asynchronous & synchronous) could lead to confusion I think. ("Do I need to call "run" after calling "start"?)

Reply:
I agree regarding the interface to keep it simple now.

My initial idea was to create a hidden run command that only runs the daemon to keep responsibilities clear. This looks a little bit like a code smell because a boolean variable is used to switch between to different inner logics (running daemon, showing help page). Or in other words: if the flag is present it behaves like a group, otherwise like a command.

I'm not sure if cobra supports such hidden commands. Otherwise we could also keep it like that for now.